src/middlewared/middlewared/plugins/domain_controller.py

```diff
@@ -21,6 +24,63 @@ class Config:
         domaincontroller['role'] = domaincontroller['role'].lower()
         return domaincontroller
 
+    @private
+    def is_provisioned(self):
+        """
+        Presumption that the domain is provisioned is to fail safe.
+        Provisioning on top of an existing domain is a destructive process that
+        must be avoided.
+        """
+        provisioned = 'org.ix.activedirectory:provisioned'
+        systemdataset = self.middleware.call_sync('systemdataset.config')
+        sysvol_path = f"{systemdataset['basename']}/samba4"
+        zfs = self.middleware.call_sync('zfs.dataset.query', [('id', '=', sysvol_path)])
+        if provisioned not in zfs[0]['properties']:
+            return False
+
+        provision_status = zfs[0]['properties']['org.ix.activedirectory:provisioned']
+
+        return False if provision_status['value'] == 'no' else True
+
+    @private
+    def set_provisioned(self, value=True):
+        systemdataset = self.middleware.call_sync('systemdataset.config')
+        sysvol_path = f"{systemdataset['basename']}/samba4"
+        ds = {'properties': {'org.ix.activedirectory:provisioned': {'value': 'yes' if value else 'no'}}}
+        self.middleware.call_sync('zfs.dataset.do_update', sysvol_path, ds)
```

Comment:
do_update is an implementation detail, for the calls just use `.update` and `.create`